draft-pda-protocol.md

```diff
@@ -437,6 +434,33 @@ Time uint64; /* seconds elapsed since start of UNIX epoch */
 * `proto`: The PPM protocol, e.g., Prio or Hits. The rest of the structure
   contains the protocol specific parameters.
 
+An `Aggregator` is structured as follows:
+
+~~~
+struct {
+  AggregatorId id;
+  Url endpoint;
+} Aggregator;
+
+uint8 AggregatorId;
+
+AggregatorId leader_aggregator_id = 0;
+
+opaque Url<1..2^16-1>;
+~~~
+
+* `id`: An identifier for the aggregator, used in HPKE encryption and which
+  allows participants to resolve entries in `Param.aggregators` against values
+  in vectors in other protocol messages. The same `AggregatorId`s MUST be used
+  when consturcting other messages containing vectors with one value for each
+  aggregator (e.g., `Report.encrypted_input_shares` in {{uploading-reports}}).
+  Each aggregator knows its own `id`. `id` values must be unique in a `Param`.
```

Comment:
Why do aggregators need to know their ID?

Reply:
```suggestion
  Each aggregator knows its own `id`. `id` values must be unique in a `Param` but may have different meanings in different Params.
```

Reply:
> Why do aggregators need to know their ID?

This is a fair question -- it seems like the aggregator ID is only used to coordinate messages, and might not need to be included in the AAD.

Reply:
Yes, an aggregator only needs to know its own ID to put in the AAD when establishing HPKE context. That is also why I made that field `uint8`, to match the previous leader/helper `server_role` values. However we could also make the aggregator ID into a longer string of bytes, allowing the values to be human readable strings, which I think will be nice for logging purposes. But it sounds like you are doubting the value of including server role in the AAD at all? I would think that since a given aggregator could be acting as either a leader or a helper in different protocol instantiations, and maybe using the same HPKE config for both, including server role diversifies keys across protocol instantiations. Then again, if that's the goal, then we should probably also mix in the `Param.uuid` since `aggregator_id` values could be re-used across different protocol instantiations.

So my take here is we should:

* define `opaque AggregatorId<1..256>;`
* use `enc, context = SetupBase(S|R)(pk, "pda input share" || Param.uuid || aggregator_id)` and `enc, context = SetupBaseS(pk, "pda output share" || Param.uuid || aggregator_id)`

But I will defer to the superior crypto brains of @chris-wood and @ekr ... should we just drop `server_role`/`aggregator_id` from the HPKE application info?

Reply:
Or we could go back to leader = `0x00` / helper = `0x01` in the info, which has the virtue of not entangling our mechanism for coordinating messages with our mechanism for key diversity, but then we will have to revisit the question of what application info should be provided by helpers beyond the first one, if and when we introduce multiple helpers. My goal in introducing the aggregator_id into the AAD was to avoid such a disruptive change.

Reply:
FWIW, I'd drop it from the AAD, and then mark an open issue asking whether or not we believe it's important to authenticate. (My impression is that this is not important to authenticate.)